Cookbook/Proposals/Comments.md

```diff
@@ -0,0 +1,223 @@
+
+I think comments are rather bad than good because of the 2 following reasons: 
+
+1. Comments eventually start to lie.
+2. Usage of comments gives a false-positive feeling to the author that the code is readable.
+
+Look at this comment:
+```swift
+/// consultation method, will be either 'phone' or 'video'
+public let consultationMethod: ConsultationMethod
+```
+
+Few lines below there's the enum declaration:
+```swift
+public enum ConsultationMethod: String {
+    case phone
+    case video
+    case faceToFace = "face to face"
+}
+```
+At the beginning, the `ConsultationMethod` had indeed 2 cases. This may be trivial here without big consequences however that not always a case and a lie may cost us many of hours.
+
+"Cross-files" comments are even more dangerous to become a lie at some point. Consider the following comment:
+```
+/// Make the booking confirmation screen. The created screen would always refetch
+/// the appointment from the backend.
+public func makeBookingConfirmation(
+    with appointmentId: AppointmentDTO.ID,
+    presentingOn flow: Flow,
+    dismiss: @escaping () -> Void
+) -> UIViewController 
+```
+Such a comment is a leak of information from another file into Builder. Builder doesn't know how BusinessController works. A change in a BusinessController can change the behavior of given screen, while there won't be a need to even enter the builder to find out there is a comment. In the end, comment stays there, unchanged.
+
+In our Pull Requests I've seen sometimes unnecessary comments
+```swift
+extension AppointmentDTO {
+    public enum State: String, Codable {
+        /// waiting for payment
+        case pending
+        /// user failed to pay for appoitment withing 10 min after the booking
+        case timedOut = "timed_out"
+        /// canceled by user or doctor
+        case cancelled
+        /// paid by user
+        case paid
+        /// doctor finished adding notes on the portal
+        case completed
+        /// patient did not answer the doctor's call
+        case noShow = "no_show"
+    }
+}
+```
+Can be simplified without any lose in understanding what the code represents to:
+
+```swift
+extension AppointmentDTO {
+    public enum State: String, Codable {
+        /// waiting for payment
+        case waitingForPayment = "pending"
+        /// timeout == 10 min
+        case paymentTimedOut = "timed_out"
+        case cancelledByUserOrDoctor = "cancelled"
+        case paidByUser = "paid"
+        case doctorFinishedAddingNotesOnThePortal = "completed"
+        case patientDidntAnswerTheCall = "no_show"
+    }
+}
+```
+
+## Proposed solution
+
+### Our approach 
+The most important thing is to improve the definition of our approach to comments as currently it isn't said clearly. Moreover, it would be nice to have a common way how we comment our code when we are forced to add them. 
```

Comment:
```suggestion
The most important thing is to improve the definition of our approach to comments as currently it isn't clearly defined. Moreover, it would be nice to have a common way how we comment our code when we are forced to add them. 
```